src/plugins/stats.ts

```diff
@@ -55,3 +52,17 @@ module.exports = async robot => {
     return popular.sort((a, b) => b.stars - a.stars).slice(0, 10)
   }
 }
+
+interface Installation {
+  id: number,
+  account: Account
+}
+
+interface Account {
+  stars: number
+}
+
+interface Repository {
+  private: boolean,
+  stargazers_count: number
+}
```

Comment:
Is there a TypeScript project that declares types for GitHub API responses?

Reply:
Wouldn't that be part of `@octokit/rest`? Since the return values from methods are pre-defined. @gr2m I can try to help with those, since it'd be quite the task to get em all (although maybe [this might be a great start](https://transform.now.sh/json-to-ts-interface/).

Reply:
@philschatz started working it at https://github.com/octokit/rest.js/pull/732. I’m reluctant of having big JSON files describing all the REST API responses without having it some automation in place that makes sure these stay up-to-date

My current idea is to use https://github.com/gr2m/octokit-rest-routes to feed automated fixtures generation for all endpoints using https://github.com/octokit/fixtures and then use the recorded, normalised fixtures to extract types.

The challenge is that response change based on requested media type.

And at least in on case the response can differ based on the request path: https://developer.github.com/v3/git/refs/#get-a-reference. I try to work around this one on Octokit level though

Reply:
The response types in https://github.com/octokit/rest.js/pull/732 were created by using auto-generated tests and fixtures for `octokat.js` but if you want to help getting them implemented in `gr2m/octokit-rest-routes` there is some discussion started in https://github.com/gr2m/octokit-rest-routes/issues/20